include/mbgl/storage/file_source.hpp

```diff
@@ -1,33 +1,33 @@
 #ifndef MBGL_STORAGE_FILE_SOURCE
 #define MBGL_STORAGE_FILE_SOURCE
 
-#include "response.hpp"
-#include "resource.hpp"
+#include <mbgl/storage/response.hpp>
+#include <mbgl/storage/resource.hpp>
 
 #include <mbgl/util/noncopyable.hpp>
-#include <mbgl/util/util.hpp>
 
 #include <functional>
-
-typedef struct uv_loop_s uv_loop_t;
+#include <memory>
 
 namespace mbgl {
 
-class Request;
+class FileRequest : private util::noncopyable {
```

Comment:
I considered an approach like this but ultimately decided to leave it up to the `FileSource` implementation what the derived destructor does. The default implementation cancels the request, but the node implementation does not.